### DISCUSSION TITLE
Quick actions in navigation / follow

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#405] [Q&A] [answered]
Quick actions we may need in navigation / follow:

- Add favorite
- Take screenshot
- Screen lock
- ...

We can see how to offer them via extra / existing buttons.

**Comment by devemux86** [comment]
Simple implementation that leaves the screen clean: tap compass and bring up a menu of large buttons with these actions.

Alternatively, add round buttons for each action, but that will fill the screen and might need settings to show / hide them.

**Reply by kds112** [reply]
Well, short tap on the compass redirects the map north. That function should stay... I vote for additional (hidable) Buttons.

**Reply by devemux86** [reply]
> short tap on the compass redirects the map north

This only happens when we browse the map and in **manual mode**: we rotate the map with gestures.

In navigation (or follow location) the map direction is fixed: north up or rotation 2D / 3D.
So the tap on compass is free in these situations.

**Comment by umpfmpf** [comment]
I would find the compass tapping and then a menu with 4 buttons quite practical and a long press for the existing compass menu.
A menu with 4 buttons is already available when tapping and long pressing the navigation button.

**Comment by devemux86** [comment]
Screenshots are complicated during navigation without opening file browser (not suitable while driving).
As Android now restricts apps to write to public storage without user permission via Android file browser.
Seems simpler to use each device's screenshot gesture instead.

So we can start with the "Add favorite" and "Screen lock".

Also the "Add favorite" can add a favorite automatically at the 1st favorite group with date time as name.
As these should be quick actions during navigation without dialogs asking questions and storage location.

**Reply by bios4** [reply]
> add a favorite automatically at the 1st favorite group with coordinates as name

I do NOT propose to add these "wow places" simply to the very first favorites group, but to a) create either a fixed "temporary favorite group" where one can find (and move if neccessary) the added favorite, or b) provide the possibility to the user to define the name of this group on his own.

Additionally, I would appreciate to see the time (when the favorite was added) and maybe even the street name in its name.

**Reply by devemux86** [reply]
We can create a group with any name and place it first to be used in navigation's favorites.
This can be done before navigation, as afterwards everything must be done automatically.

Find current street name is not guaranteed, it can be done only **online** by converting coordinates to an address.
This is usually a snapped address on any nearby street and very often not the exact location of the coordinates.
e.g. long press in Cruiser, select "Show address" and you will see that the result is not always where we click.

Also as the favorites currently only have name, we have to think what to store in the name field.

**Reply by umpfmpf** [reply]
Perhaps the time and date as Name

**Reply by devemux86** [reply]
@bios4 @umpfmpf 

I updated my post, indeed we need the date time as name.

**Reply by 0709wiwiwi** [reply]
> Screenshots are complicated during navigation without opening file browser (not suitable while driving).
   Because Android apps are now restricted from writing to public storage without user permission via the Android file browser.

Prefer ONLY to use **Android>media>gr.talent.cruiser.** 
Put all the desired free accessible folders in here. (A13)
By action "Cruiser screenshot" > name on date_time > auto save in Cruiser Screenshots folder.
The CX file explorer (library images) will detect and show the Cruiser Screenshots folder in media.
And through the CX file explorer, easily produce shortcuts to accessible Cruiser Folders in media.

**Reply by devemux86** [reply]
"Android/media" folder is a simple solution for the screenshots.
I can add it like this if you like.

(as currently we do not have a screenshot manager like in favorites)

**Comment by bios4** [comment]
I'd like to add my vote for the "[Show dynamic route overview](https://github.com/devemux86/cruiser/discussions/217#discussioncomment-6583910)" suggestion after a long tap on the compass.

**Reply by Ciutnik** [reply]
Or maybe use a long press on the zoom-out button? Although it would be logical if it triggered a temporary route review. And then the dynamic route overview (constant) could be as an option for the map direction (according to  @WolfgangZ42 proposal) or navigation panel menu item (here is the route overview now).

**Reply by devemux86** [reply]
Route overview can be improved with more functions, it is already in my list. :slightly_smiling_face: 

We also have a discussion here:
- https://github.com/devemux86/cruiser/discussions/94

**Comment by devemux86** [comment]
The first quick actions in navigation / follow are coming in next Cruiser:
(tap on compass)

<img src="https://github.com/devemux86/cruiser/assets/3484020/e6ec7fc9-7145-45b8-89a3-0850fb9b7482" width="50%"/>

**Reply by GuG1050** [reply]
I Like that 👍

**Reply by r11d11** [reply]
I like the quick menu, too. First I had problems to find the saved favorite, because my first favourite group in my places has 400 entries. So I changed my first favourite group in my places to an empty group an so I have no problems to find my saved favorites.

**Reply by devemux86** [reply]
> First I had problems to find the saved favorite

We could somehow improve this. :slightly_smiling_face:

**Reply by airald** [reply]
POIs are saved with no prompt to first POI category. I do the same as @r11d11, created a new group called Auto POI and placed in the first position to collect all PDI taken with this shortcut. Maybe a toast message saying "saved to XXXX group" could be helpful.

**Reply by umpfmpf** [reply]
I find it very successful to operate with the quick saving of a point.

**Reply by WolfgangZ42** [reply]
> I find it very successful to operate with the quick saving of a point.

Me too! Because I only use the "quick screenshot" to correct the osm data afterwards, I can now use the position directly (without going through date time and gps recording).

**Reply by Ciutnik** [reply]
If a link could be added to the "quick point" (if description fields are implemented) so that it could be opened on the OpenStreetMap website, it would make it easier to make changes to the map 🙂

**Reply by devemux86** [reply]
@Ciutnik do you mean an OpenStreetMap link created from the favorite coordinates?

There is already the "Show on OpenStreetMap" action in the "More..." menu of all map elements.

**Reply by Ciutnik** [reply]
> There is already the "Show on OpenStreetMap" action in the "More..." menu of all map elements.

Yes, it's not a problem in the app - I do it often. But it would be nice to have this option on a desktop computer (I can easily synchronize my entire Favorites collection with any device via the cloud ). The OSM online editor is not very friendly to Android web browsers. Maybe I'm wrong, but I don't think the coordinates contained in GPX are "clickable" to easily open the OSM website?

**Comment by devemux86** [comment, marked as answer]
The first _quick actions_ in navigation / follow are available in [Cruiser 3.0.37](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

(press compass)

**Comment by bios4** [comment]
Question to @devemux86: was there a certain reason to add these two _quick actions_ to the tap-on-compass menu?

As the compass is VERY hard to hit being on the road and wearing gloves, I would like to suggest to move the two buttons to the tap-on-direction-arrow menu - see example screenshots now <> then:

<img src="https://github.com/devemux86/cruiser/assets/145460147/929a8a16-3a18-4cb5-bec8-cff12fa7507d" width="33%">
<img src="https://github.com/devemux86/cruiser/assets/145460147/aa8e700b-5e53-4097-82e5-306b6acad966" width="33%">

If you do not want to move these tiles to the direction-arrow-menu, my suggestion would be to alternatively
- increase the space between the compass icon and the panel displayed above it (the next fuel station POI in my screenshot) as well as
- increase the touch-area of the compass to make it easier to operate on the road/with gloves (same as for the [off-route warning button](https://github.com/devemux86/cruiser/discussions/482#discussioncomment-9244036))

**Reply by bios4** [reply]
OK, I just realised this was probably done because you wanted to provide these functions also in FOLLOW mode.
There is no direction icon of course. 🙄

So then please consider my proposal for the increased touch area...

**Reply by devemux86** [reply]
We included these actions when tap on compass because they're not just actions used in navigation,
but they can also be used in follow location mode, so we need a common way to use them.

- We could add them in navigation actions, but then there will be different use: navigation vs follow.
- We can also increase the touch area of the compass? (this can be done anyway)

**Reply by bios4** [reply]
Yes, I realized this right before your answer ^^...

> We could add them in navigation actions, but then there will be different use: navigation vs follow.

Another suggestion: display an explicit "quick actions" panel in the same size and on the same position as the direction-arrow panel when follow mode is active!?

Means during navigation, the direction-arrow-menu contains all the tiles shown in my mockup.
When switched to follow mode (after arrival) or during follow mode in general, the direction-arrow-panel is replaced by a "quick actions" panel containing only the (currently two) quick actions...

**Reply by devemux86** [reply]
Follow after navigation is also a case, the way of user actions better not change, it will be confusing.

An extra panel filling the screen for actions seems unnecessary when the compass is already available?
Let's start by increasing the touch area of the compass to see how it goes.
(we already did a similar thing in the off route warning button)

**Reply by Ciutnik** [reply]
Yes, it is true that it is difficult to point a compass with a gloved finger. That's why, quite recently, I suggested that the compass should be the same size as the buttons. 
Everyone can set it depending on the size of the screen and... their fingers 🙂

**Reply by devemux86** [reply]
> That's why, quite recently, I suggested that the compass should be the same size as the buttons.

Let's try this.

**Reply by bios4** [reply]
> > That's why, quite recently, I suggested that the compass should be the same size as the buttons.
> 
> Let's try this.

Is this still planned?

In 4.1.0-b2 the compass size is not yet affected by the button size setting...

**Reply by devemux86** [reply]
> the compass should be the same size as the buttons

The compass is now the same size as the buttons (default size).

> the compass size is not yet affected by the button size setting

This is not available yet, if we need it I can implement it.

**Reply by bios4** [reply]
👍 - I think this would make sense...

**Comment by devemux86** [comment]
Next Cruiser will also change the compass size with the "Setttings | Application | Appearance | Button size".

**Comment by MotoTraveller** [comment]
**Screen lock**

good idea since my displays going mad when it starts raining.
I have two question.

1. once screen lock has been activated, what is the thought way to turn it off again? - I just found the screen lock killing process by wiping the android menu down. Is this the way it should be done or is there another locking escape process?
2. unfortunately, the screen lock at my android covers only the area of the cruiser App but not the 2 control lines on the screen for the top and bottom wiping/clicking operations. Is it possible to have the lock all over the whole screen? 

Best regards.

**Reply by devemux86** [reply]
@MotoTraveller 

1. When screen lock is activated a permanent notification appears.
You can use this notification to disable the screen lock.

2. Apps can only control their screens, not the whole Android system.
You can enable the full screen mode in app settings, so the app fills the screen.

**Reply by MotoTraveller** [reply]
Hi thanks for feedback:

to 2: Ok - I understand

to 1: Well, I have no permanent notification. Once I activate it, a short info icon is popping up (see screenshot) that it has been activated but it disappears after 1 or 2 seconds. However I tried to be fast to click on it while it appears but nothing happened screen was still locked and I could turn the screen lock only off by wiping screen down and deactivate the locking process there (see 2nd screenshot)

<img width="800" height="1280" alt="image" src="https://github.com/user-attachments/assets/858973f3-de11-4009-887b-fb22f8cddc37" />


<img width="796" height="970" alt="image" src="https://github.com/user-attachments/assets/6a67e325-f772-44fa-929e-a5fdfa648582" />

**Reply by devemux86** [reply]
>  I could turn the screen lock only off by wiping screen down and deactivate the locking process there (see 2nd screenshot)

@MotoTraveller This is the permanent screen lock notification and how to disable it. :slightly_smiling_face: 

(notifications on Android appear at the top)

**Reply by MotoTraveller** [reply]
Yes, but on your first answer: "_When screen lock is activated a permanent notification appears.
You can use this notification to disable the screen lock."_ 

I have no permanent notification for display lock which remains on screen while it's locked - I'm using an android tablet for navigation.

**Reply by devemux86** [reply]
@MotoTraveller 

Notifications on Android appear at the top.
In your 2nd screenshot you see the notifications.
The 1st notification appears when the screen lock is activated.

**Reply by MotoTraveller** [reply]
Ahhh, you mean this small icon (in the top line) that shows that display is locked?
<img width="800" height="553" alt="image" src="https://github.com/user-attachments/assets/3467a94c-434d-4f5f-89ac-810721d98e06" />

Then I was write and deactivation is only by swiping down and turn it off like the way I found allready.
I understood it in that way that when you wrote "on top" it should be there as a permanent icon layer like on top over the application.

**Reply by devemux86** [reply]
@MotoTraveller Correctly, the top location of the screen where all Android notifications appear.
Like the updates from Google Play (your 3rd notification).
